pcdsdevices/analog_signals.py

```diff
@@ -61,6 +61,33 @@ class Acromag(BaseInterface, Device):
     tab_component_names = True
 
 
+def acromag_factory_func(prefix, channel, *, signal_class=None, name=None,
+                         **kwargs):
+    """
+    This is a factory function for creating an Acromag output or input signal.
+
+    Parameters
+    ----------
+    prefix : str
+        The base EPICS PV for the Acromag. E.g.: `XPP:USR:ai1`
+    channel : int or str
+        The channel number. E.g.: `1` [0-15]
+    signal_class : str, optional
+        Signal Class to use. E.g.: `EpicsSignalRO`, `EpicsSignal`. If `None`
+        provided, it will use the prefix to try to generate the appropriate
+        signal class.
+    name : str, optional
+        A name to refer to the Acromag Channel, if `None` it will use the
+        prefix and channel to generate a name. E.g.: `ai_1` or `ao_1`.
+    """
+    if signal_class is None:
+        signal_class = EpicsSignalRO if ':ai' in prefix else EpicsSignal
+    name_prefix = 'ai_' if ':ai' in prefix else 'ao_'
+    name = name or ''.join([name_prefix, str(channel)])
+    prefix = ''.join([prefix, ':', str(channel)])
+    return signal_class(prefix, name=name, kind='normal')
```

Comment:
Is this kind of what we want?

Reply:
Looks good, I think! You might consider f-strings for the `join` calls, but you don't need to.

Reply:
Resolved in 9290102